### PR TITLE
qa/rgw: pull Apache artifacts from mirror instead of archive.apache.org

### PR DESCRIPTION
**Pull request by ivancich**
Currently maven and kafka are pulled from archive.apache.org. This uses Apache's "closer" calculator to find a mirror to use instead.

Fixes: https://tracker.ceph.com/issues/69235

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by ivancich** [comment]
QA -- hadoop and kafka tests no longer fail due to inaccessible artifacts from Apache. One hadoop job fails for a different reason.

https://pulpito.ceph.com/ivancich-2024-12-14_01:29:16-rgw-wip-eric-testing-2-distro-default-smithi/

**Comment by ivancich** [comment]
jenkins test api

File: qa/suites/rgw/notifications/tasks/kafka/test_kafka.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 tasks:
 - kafka:
     client.0:
-      kafka_version: 2.6.0
+      kafka_version: 3.8.1
```
Comment (cbodley) [inline qa/suites/rgw/notifications/tasks/kafka/test_kafka.yaml:4]: cc @yuvalif

**Review by cbodley — APPROVED** [review]
lgtm

File: qa/suites/rgw/notifications/tasks/kafka/test_kafka.yaml (hunk 1 above)
Comment (yuvalif) [reply qa/suites/rgw/notifications/tasks/kafka/test_kafka.yaml:4]: ran basic sanity with 3.8.0 - works ok

Comment (ivancich) [reply qa/suites/rgw/notifications/tasks/kafka/test_kafka.yaml:4]: The mirrors don't seem to go back to 2.6.0, so I pulled it up.

**Pull request merged**